tests/macros/groupdata.test.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * @prettier
+ */
+
+const { itMacro, describeMacro } = require('./utils');
+
+/*
+Different strings in GroupData objects have different sets of
+permitted characters.
+*/
+const permittedCharacters = {
+  group:      /^[\w ()-]+$/,
+  overview:   /^[\w ()-]+$/,
+  interface:  /^[\w.]+$/,
+  property:   /^[\w.]+$/,
```

Comment:
Similarly, property and method names always start with a lower-case letter (except for constructors).

Reply:
There are many cases where this isn't true in the data, e.g. https://github.com/mdn/kumascript/blob/master/macros/GroupData.json#L6.

We could make the validation a bunch more complex here (split on dots, and validate the pieces separately) but I don't think it's worth it. The main point of this test is to make sure that assumptions made by the macros aren't broken by data updates.

Reply:
Fair enough. I was thinking we could protect against certain typos, too, but you're right that that's outside the scope of verifying the structure of the data.